### DISCUSSION TITLE
derivative not able to read change in status of a cell

### DISCUSSION THREAD

**Discussion by shaunnthomas** [AntlerVC/firetable#1492] [Support / Q&A] [answered]
I have added a derivative column to a row that listense to changes in another column called 'status' when the status is 'requested' it should run the function. Here is a snippet of my code: 
`const derivative: Derivative = async ({row, db}) => {
  // Check if the status is 'requested'
  if (row.status === 'requested') {
    // Get the API key or authentication token from Rowy secrets
    const apiSecret = await rowy.secrets.get("assemblyai");

    // Prepare the transcription request payload
    const transcriptionPayload = {
      audio_url: row.fileURL,
      language_code: row.language,
      speaker_labels: row.identifySpeakers, // Set the language code accordingly
      punctuate: true,

      // Add other parameters as needed based on your requirements
    };

    // Make the API request to initiate transcription
    const response = await fetch("https://api.transcription-service.com/v2/transcript", {
      method: "POST",
      headers: {
        Authorization: `Bearer ${apiSecret}`,
        "Content-Type": "application/json",
      },
      body: JSON.stringify(transcriptionPayload),
    });

    if (response.ok) {
      // Extract the transcript ID from the response
      const { id } = await response.json();

      // Store the transcript ID in the Firestore document
        await db.collection('transcriptRequest').doc(row.id).update({ transcriptId: id });

      // Return any relevant information or perform additional actions if needed
      return { success: true, message: `Transcription initiated with ID: ${id}` };
    } else {
      // Handle API request error
      return { success: false, message: `Error initiating transcription: ${response.statusText}` };
    }
  } else {
    // If the status is not 'requested', do nothing
    return { success: true, message: 'Transcription not requested' };
  }
};

export default derivative;
`
Is there something wrong with my code? When I change the status to something other than 'requested' the function does run with a response of { success: true, message: 'Transcription not requested' };

**Comment by Chadha93** [comment, marked as answer]
Hi @shaunnthomas 

One possible reason for this behavior could be that the `row.status` value is not exactly 'requested' when you are expecting it to be. It is possible that there are leading or trailing spaces in the `row.status` value, or the case of the string is different. 

To ensure that the condition is properly checked, you can modify your code to trim the `row.status` value and convert it to lowercase before checking the condition. 

Here's an updated version of your code with the modification:

```javascript
const derivative: Derivative = async ({ row, db }) => {
  // Trim and convert row.status to lowercase
  const status = row.status.trim().toLowerCase();

  // Check if the status is 'requested'
  if (status === 'requested') {
    // Rest of your code
    // ...
  } else {
    // If the status is not 'requested', do nothing
    return { success: true, message: 'Transcription not requested' };
  }
};

export default derivative;
```

By trimming the `row.status` value and converting it to lowercase, you can ensure that any leading/trailing spaces or case differences are eliminated, and the condition is correctly checked.

**Reply by Chadha93** [reply]
Marking above as an answer, feel free to re-open or followup.
